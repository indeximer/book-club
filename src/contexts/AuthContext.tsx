"use client";
import { useRouter } from "next/navigation";
import {
  createContext,
  ReactNode,
  useContext,
  useEffect,
  useState,
} from "react";
import {
  onAuthStateChanged,
  signInWithEmailAndPassword,
  signOut,
  createUserWithEmailAndPassword,
  updateProfile,
} from "firebase/auth";
import { auth } from "@/config/firebase";

type AuthProps = {
  user: {} | null;
  signIn: (email: string, password: string) => void;
  signUp: (name: string, email: string, password: string) => void;
  logOut: () => void;
};

const AuthContext = createContext<AuthProps>({
  user: null,
  signIn: (email: string, password: string) => undefined,
  signUp: (name: string, email: string, password: string) => undefined,
  logOut: () => undefined,
});

export function AuthProvider({ children }: { children: ReactNode }) {
  const [user, setUser] = useState<any>(null);
  const router = useRouter();

  useEffect(() => {
    const unsubscribe = onAuthStateChanged(auth, (user) => {
      setUser(user);
    });
    return () => unsubscribe();
  }, []);

  const signIn = (email: string, password: string) => {
    return signInWithEmailAndPassword(auth, email, password);
  };

  const logOut = () => {
    return signOut(auth);
  };

  const signUp = async (name: string, email: string, password: string) => {
    const userResponse = await createUserWithEmailAndPassword(
      auth,
      email,
      password
    );
    console.log(user);
    updateProfile(auth.currentUser!, {
      displayName: name,
    });
    router.push("/");
  };

  return (
    <AuthContext.Provider value={{ user, signIn, signUp, logOut }}>
      {children}
    </AuthContext.Provider>
  );
}

export function useAuth() {
  return useContext(AuthContext);
}
