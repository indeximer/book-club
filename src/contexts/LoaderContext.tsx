"use client";

import { Splash } from "@/components/splash";
import { createContext, ReactNode, useContext, useState } from "react";

type LoaderProps = {
  setLoading: (loading: boolean) => void;
};

const LoaderContext = createContext<LoaderProps>({
  setLoading: (loading) => undefined,
});

export function LoaderProvider({ children }: { children: ReactNode }) {
  const [loading, setLoading] = useState(false);

  return (
    <LoaderContext.Provider value={{ setLoading }}>
      {loading && <Splash />}
      {children}
    </LoaderContext.Provider>
  );
}

export function useLoader() {
  return useContext(LoaderContext);
}
