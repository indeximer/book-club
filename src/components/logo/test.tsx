import { render } from "@testing-library/react";
import { Logo } from ".";
import { AuthProvider } from "@/contexts/AuthContext";
import { LoaderProvider } from "@/contexts/LoaderContext";

jest.mock("next/navigation", () => ({
  useRouter() {
    return {
      prefetch: () => null,
    };
  },
}));

describe("<Logo />", () => {
  it("should render without breaking", () => {
    const { container } = render(
      <LoaderProvider>
        <AuthProvider>
          <Logo />
        </AuthProvider>
      </LoaderProvider>
    );

    expect(container.firstChild).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(
      <LoaderProvider>
        <AuthProvider>
          <Logo />
        </AuthProvider>
      </LoaderProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
