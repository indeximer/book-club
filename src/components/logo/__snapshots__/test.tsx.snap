// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should match snapshot 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-4quli5-MuiTypography-root"
    href="/"
  >
    BookClub
  </a>
</div>
`;
